packages/flutter/lib/src/widgets/default_text_editing_actions.dart

```diff
@@ -29,10 +31,54 @@ class DefaultTextEditingActions extends Actions{
     required Widget child,
   }) : super(
     key: key,
-    actions: _shortcutsActions,
+    actions: <Type, Action<Intent>>{
+      ..._shortcutsActions,
+      ..._platformActions,
+    },
     child: child,
   );
 
+  static final Map<Type, Action<Intent>> _androidActions = <Type, Action<Intent>>{
+    SingleTapUpTextIntent: _SelectLastTapDownPositionTextAction(),
```

Comment:
I know you mentioned it in a previous message, but is there further context on why we want to intents to be the events and no named after the actual thing that needs to be done? Why is it done one way with Shortcuts and another way with Gestures?

I understand we need to maintain a mapping somewhere but what makes sense in my mental model of Actions/Intents is:
- Gesture is triggered in the text editing gesture detector
- A parent widget translates those gestures to intents (can maintain a default mapping for all platforms)
- The action executes the command on the renderEditable

Otherwise, I feel like we're treating an intent differently based on the situation. It has a different meaning for gestures than it does for shortcuts.

I may confusing what the difference between an intent and action is though.

Reply:
When you support triple click select all text by Desktop ?

Reply:
@letungcntt There are plans for @venkatd to take a look at triple click in the future.  This PR will not do that yet.

Reply:
@venkatd Here's some context from when the naming decision was made: https://github.com/flutter/flutter/pull/75032#discussion_r570518251

Basically my reasoning is separation of concerns.  I don't want visual text editing widgets like TextField to be concerned with text editing business logic.  I want to define text editing business logic exclusively in DefaultTextEditingActions and have it apply everywhere.  Then, widgets like TextField are only concerned with visuals and detecting gestures.

Reply:
@justinmc thanks for sharing!

I understand the goal and read through the discussions in the google doc. I can't help but think that, if intents have this dual nature, if there's something that needs to be resolved at the Shortcuts/Intents/Actions abstraction layer.

Or perhaps it's just a matter of how Intents/Actions are named.

Either way, it's probably out of the scope of what you're working on. I'll stick with whatever structure you have in this PR.

Reply:
@gspencergoog I want to resurrect the discussion about Shortcut/Action/Intent naming to talk about an idea that @venkatd [posted in the design doc](https://docs.google.com/document/d/1QaVIr1bbOWJGNyyNsB75sXgTDRMuxP268sLGTBNDut0/edit?disco=AAAAIOJhwOE).  The idea is to create a superclass of Shortcuts (we'll call it Events) which also trigger Intents, but which can be invoked directly instead of being tied to keyboard shortcuts.

### Background

Currently, this PR does things according to our original decision, where we decided to stick with the standard naming for keyboard stuff:

 * Shortcut: What keys were pressed. `LogicalKeySet(LogicalKeyboardKey.arrowLeft)`
 * Intent: What we intend to do. `MoveSelectionLeftTextIntent`
 * Action: What it does. `_MoveSelectionLeftTextAction`

But for things invoked directly with Actions.invoke, like taps, we would name the Intent after what happened:

 * Shortcut: Not used.
 * Intent: What happened. `SingleTapUpTextIntent`
 * Action: What it does: `_SelectLastTapDownPositionTextAction`

The reason that I advocated for names like SingleTapUpTextIntent is so that classes like TextSelectionGestureDetectorBuilder can be ignorant of what the gestures it detects actually do, but you can see that Intents are used in two different ways, which is still a bit confusing.

### New idea

The new idea would keep keyboard shortcuts the same, but change taps to work like this:

 * Event: What happened. `SingleTapUpTextEvent`
 * Intent: What we intend to do. `SelectLastTapDownPositionTextIntent`
 * Action: What it does: `_SelectLastTapDownPositionTextAction`

With this way, the whole system is used in the same way for both keyboard and taps.  Rather than separating shortcutActions and platformActions in DefaultTextEditingActions, the platform mapping for taps would happen at the DefaultTextEditingEvents level in the same way that it happens for keyboard shortcuts in DefaultTextEditingShortcuts.

Do you think this is a viable solution that is in line with how Shortcuts/Intents/Actions are intended to be used?

Reply:
I did intend that it would be a widget, because I need a place to map the Triggers (let's call them) to Intents, and it needs to be overridable by users and all of that.

So in this case:

  * TextSelectionGestureDetectorBuilder calls `Triggers.invoke<SingleTapUpTextTrigger>(...`
  * DefaultTextEditingTriggers has mapped SingleTapUpTextTrigger to SelectLastTapDownPositionTextIntent and SelectLastTapDownPositionOrEdgeTextIntent based on the platform, and it invokes the right one.
  * DefaultTextEditingActions has mapped SelectLastTapDownPositionTextIntent to _SelectLastTapDownPositionTextAction (and similar for the other Intent).

Is that adding too much complexity since we already have Shortcuts?

Reply:
You could just add a `triggers` map to the `Shortcuts` widget, and have it do double duty, but I guess that's pretty confusing naming-wise.

Should it really be a base class to `Shortcuts`, or should they just share a common base? For instance, I'm wondering `Shortcuts` sets/uses the `triggers` map on the `Triggers` class if it is the base for `Shortcuts`. They seem more like siblings with common functionality.

In any case, those are all implementation details.  I think it seems like a reasonable idea.

Reply:
Makes sense that the word `Event` is too overloaded to have a clear meaning.

Perhaps for now there could be a parallel `Triggers` abstraction that can be treated as an implementation detail? That could allow us to compare differences with `Shortcuts` to decide if generalization makes sense, and later if a public API would be helpful.